src/psyclone/psyir/backend/sympy_writer.py

```diff
@@ -418,7 +447,7 @@ def __call__(self, list_of_expressions):
         is_list = isinstance(list_of_expressions, (tuple, list))
```

Comment:
Since this is the entry point at which the caller first supplies `assumes`, we should do some type-checking here.